Description: Add type signatures to allow building against parsec3
Author: Joachim Breitner <nomeata@debian.org>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- washngo-2.12.0.1.orig/WashNGo.cabal
+++ washngo-2.12.0.1/WashNGo.cabal
@@ -21,7 +21,7 @@ Cabal-Version: >= 1.2
 --  location: http://code.haskell.org/WASH/

 Library
-  Build-Depends: base < 5, regex-compat, haskell98, parsec < 3
+  Build-Depends: base < 5, regex-compat, haskell98, parsec >= 3
   Extensions: ForeignFunctionInterface
   build-depends: containers
   Exposed-Modules:
--- washngo-2.12.0.1.orig/WASH/Mail/MailParser.hs
+++ washngo-2.12.0.1/WASH/Mail/MailParser.hs
@@ -38,6 +38,7 @@ literalString = do char '\"'
		   return str

 no_ws_ctl_chars = map chr ([1..8] ++ [11,12] ++ [14..31] ++ [127])
+no_ws_ctl :: Parser Char
 no_ws_ctl = oneOf no_ws_ctl_chars

 text_chars = map chr ([1..9] ++ [11,12] ++ [14..127])
--- washngo-2.12.0.1.orig/WASH/Mail/RFC2822.hs
+++ washngo-2.12.0.1/WASH/Mail/RFC2822.hs
@@ -14,8 +14,11 @@ fws =
   do crLf
      many1 ws1

+ws1 :: Parser Char
 ws1 = oneOf " \t"
+lineChar :: Parser Char
 lineChar = noneOf "\n\r"
+headerNameChar :: Parser Char
 headerNameChar = noneOf "\n\r:"

 -- |parse contents of Date field according to RFC2822
--- washngo-2.12.0.1.orig/washparser/hs/WASHParser.hs
+++ washngo-2.12.0.1/washparser/hs/WASHParser.hs
@@ -49,6 +49,7 @@ xmlfile = do {


 -- 2.2 / 2
+char' :: Parser Char;
 char' = (char '\t' <|> char '\n' <|> char '\r' <|>
     satisfy (>= ' ')) <?> "character";
